amulet/sentry.py

```diff
@@ -254,24 +261,26 @@ def wait_for_status(self, juju_env, services, timeout=300):
             raise
 
     def wait(self, timeout=300):
-        #for unit in self.unit:
         ready = False
         try:
             with helpers.timeout(timeout):
                 # Make sure we're in a 'started' state across the board
-                waiter.wait(timeout=timeout)
-                while not ready:
+                now = time.time()
+                while not ready and now + timeout < time.time():
```

Comment:
I'm not sure you want the extra time checking here. Clients are expecting this method (wait()) to raise if units aren't ready within the timeout. I think just `while not ready` is better here.